### PR TITLE
Math interpolants and remainings: Move to es6 classes

### PR DESCRIPTION
**Pull request by yomotsu**
A part of #19984.
A series of Math classes conversion, Part 3 of 3.

This PR contains:

- ~[ ] Interpolant (The base class)~
  - [ ] CubicInterpolant
  - [ ] DiscreteInterpolant
  - [ ] LinearInterpolant
  - [ ] QuaternionLinearInterpolant
- ~[ ] **Interpolant.tests.js** [*1]~

and
- [x] Color

---

~*1:
The test has to be changed as well because 'Interpolant.tests' was assuming `Interpolant` base class to be a Function (using `call` to extend the base class).~

---

edited: `Interpolant` base class [couldn't be converted](https://github.com/mrdoob/three.js/pull/20086#issuecomment-674514866). I keep it as it is.

### CODE REVIEW

File: src/math/Interpolant.js [hunk 1]
```diff
@@ -191,22 +191,15 @@ Object.assign( Interpolant.prototype, {
...
 		return this.interpolate_( i1, t0, t, t1 );
 
-	},
-
-	settings: null, // optional, subclass-specific settings structure
-	// Note: The indirection allows central control of many interpolants.
-
-	// --- Protected interface
-
-	DefaultSettings_: {},
```
Comment (yomotsu) [inline src/math/Interpolant.js:201]: `settings` and `DefaultSettings_` is moved to the end of the file.

File: test/unit/src/math/Interpolant.tests.js [hunk 2]
```diff
@@ -9,71 +9,73 @@ export default QUnit.module( 'Maths', () => {
 		// Since this is an abstract base class, we have to make it concrete in order
 		// to QUnit.test its functionality...
 
-		function Mock( parameterPositions, sampleValues, sampleSize, resultBuffer ) {
+		class Mock extends Interpolant {
 
-			Interpolant.call( this, parameterPositions, sampleValues, sampleSize, resultBuffer );
```
Comment (yomotsu) [inline test/unit/src/math/Interpolant.tests.js:14]: `Interpolant` abstract class is not a function anymore. the test is also needed to change.

**Comment by mrdoob** [comment]
Seems like `webgl_loader_mmd` broke.

**Comment by yomotsu** [comment]
Aww...
It because MMDLoader also assumes Interporant to be a function...
https://github.com/mrdoob/three.js/blob/4c6879aa668a9f7aa7a89eb461aec24028290cd8/examples/js/loaders/MMDLoader.js#L1840

Should we keep the Interporant base class as a function? It may break other 3rd partiy implementations

**Comment by Mugen87** [comment]
Notice that `GLTFCubicSplineInterpolant ` also derives from `Interpolant`.

https://github.com/mrdoob/three.js/blob/908de674229c5096d52747df17e2da16df3ddfee/examples/jsm/loaders/GLTFLoader.js#L1218-L1222

**Comment by yomotsu** [comment]
Thanks for your notice. I'll revert it for now.

**Comment by yomotsu** [comment]
let me close this PR.
I will open another PR for Color class.

**Pull request closed without merging**